examples/preprocessing/plot_artifact_correction_with_faster.py

```diff
@@ -0,0 +1,119 @@
+event_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw-eve.fif'
+
+raw = io.Raw(raw_fname, preload=True)
+raw.info['bads'] = []  # bads are going to be detected automatically
+events = mne.read_events(event_fname)
+
+# Keep whatever EEG reference the amplifier used for now. After the data is
+# cleaned, we will re-reference to an average reference.
+raw, _ = io.set_eeg_reference(raw, [])
+
+# Highpass filter the EEG, EOG and ECG data to eliminate drifts
+# NOTE: we do not lowpass filter here, because power line noise is useful for
+#       detecting bad channels.
+picks = mne.pick_types(raw.info, meg=False, eeg=True, eog=True, ecg=True)
+raw.filter(0.3, None, method='iir', picks=picks)
+
+# Construct epochs
+event_ids = {'AudL': 1, 'AudR': 2, 'VisL': 3, 'VisR': 4}
+tmin = -0.2
+tmax = 0.5
+picks = mne.pick_types(raw.info, meg=True, eeg=True, eog=True, ecg=True)
+epochs = mne.Epochs(raw, events, event_ids, tmin, tmax, baseline=(None, 0),
+                    preload=True, picks=picks)
+
+# Compute evoked before cleaning, using an average EEG reference
+epochs_before = epochs.copy()
+epochs_before.info['custom_ref_applied'] = False
+epochs_before, _ = io.set_eeg_reference(epochs_before)
+epochs_before.apply_proj()
+evoked_before = epochs_before.average()
+
+###############################################################################
+# Clean the data using FASTER
+
+# Step 1: mark bad channels
+eeg_picks = mne.pick_types(epochs.info, meg=False, eeg=True, exclude=[])
+epochs.info['bads'] = faster_bad_channels(epochs, picks=eeg_picks, thres=5)
+
+meg_picks = mne.pick_types(epochs.info, meg=True, exclude=[])
+epochs.info['bads'] += faster_bad_channels(epochs, picks=meg_picks, thres=5)
+
+# Step 2: mark bad epochs
+
+# Picks have to be re-done to exclude the bads we just detected
+eeg_picks = mne.pick_types(epochs.info, meg=False, eeg=True, exclude='bads')
+bad_epochs = faster_bad_epochs(epochs, picks=eeg_picks, thres=3)
+
+meg_picks = mne.pick_types(epochs.info, meg=True, exclude='bads')
+bad_epochs += faster_bad_epochs(epochs, picks=meg_picks, thres=3)
+
+# Reject bad epochs (keep good epochs)
+good_epochs = list(set(range(len(epochs))).difference(set(bad_epochs)))
+epochs = epochs[good_epochs]
+
+# Step 3: mark bad ICA components and remove them
+picks = mne.pick_types(epochs.info, meg=True, eeg=True, eog=True, ecg=True,
+                       exclude='bads')
+ica = ICA(n_components=0.95).fit(epochs, picks=picks)
+
+# generate EOG and ECG epochs to improve detection by correlation
+raw.info['bads'] = epochs.info['bads']
+picks = mne.pick_types(raw.info, meg=True, eeg=True, eog=True, ecg=True,
+                       exclude='bads')
+eog_epochs = create_eog_epochs(raw, picks=picks)
+ecg_epochs = create_ecg_epochs(raw, picks=picks)
+eog_components, _ = ica.find_bads_eog(eog_epochs)
+ecg_components, _ = ica.find_bads_ecg(ecg_epochs)
+ica.exclude = eog_components + ecg_components
```

Comment:
It's sort of implemented as `faster_bad_components`, but I have a feeling the existing ICA artifact detection/rejection procedures are superior to FASTER.